examples/kustomization/sts-example/sample-data/mc-job-setup-bucket.yaml

```
@@ -42,17 +42,37 @@ spec:
           configMap:
             name: start-config-script
             defaultMode: 0744
+        - name: sa-token
+          projected:
+            sources:
+              - serviceAccountToken:
+                  audience: "sts.min.io"
+                  expirationSeconds: 86400
```

86400 is 1 day, and is a little long. I think AWS uses it because they are out-of-cluster.

The Kubernetes default 1 hour (720 seconds). I would stay with the default or something closer to it rather than a long period like 1 day.

```suggestion
                  expirationSeconds: 720
```

---

Before expiration, Kubernetes will automatically replace the token with a new refreshed token with an extended life time. As long as the application is reloading new secrets from disk, they should never observe an expired token.

I'm pretty sure the AWS SDKs using `AWS_WEB_IDENTITY_TOKEN_FILE` should re-read the updated file, as this is normal behavior for IRSA. It may be worth double checking `mc` or Minio client SDKs though to make sure though.

---

I somehow failed to do math, I updated the suggestion above to be 3600.

---

MinIO SDK's and Operator reads the token from the file allways, 1 hour should be enough